ecleankernel/__main__.py

```diff
@@ -256,6 +261,34 @@ def main(argv: typing.List[str]) -> int:
                 bootloader=bootloader,
                 destructive=args.destructive)
 
+            if HOOK_DIR.is_dir():
+                preserved_kernels = [k for k in kernels
+                                     if k not in removals]
+
+                def as_version_list(kernels: List[Kernel]) -> str:
+                    return " ".join(list(map(lambda x: x.version, kernels)))
```

Comment:
Why the `list` conversion? Also comprehension would probably be shorter than `map()`.